### PR TITLE
fix stretch effect with rtl support

### PR DESCRIPTION
**Pull request by youssefali424**
fix for #113185

### CODE REVIEW

**Comment by goderbauer** [comment]
@youssefali424 Thanks for the contribution. It looks like it is missing a test, though, and there are also some failing checks. Could you add a test and take a look at the failing tests?

/cc @Piinks for when this is ready for review.

**Comment by youssefali424** [comment]
@Piinks please review

File: packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart [hunk 1]
```diff
@@ -218,6 +223,92 @@ void main() {
...
+    );
+
+    TestGesture gesture = await tester.startGesture(tester.getCenter(find.byType(CustomScrollView)));
+    // Overscroll the start
+    await gesture.moveBy(const Offset(200.0, 0.0));
+    await tester.pumpAndSettle();
+
+    // expect(box1.localToGlobal(Offset.zero), Offset.zero);
+    // expect(box2.localToGlobal(Offset.zero).dx, greaterThan(305.0));
+    // expect(box3.localToGlobal(Offset.zero).dx, greaterThan(610.0));
```
Comment (Piinks) [inline packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart:265]: Should these be removed?

**Review by Piinks — COMMENTED** [review]
Hey @youssefali424 thanks for fixing this! Can you take a look at some of the commented out code in the test?

File: packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart (hunk 1 above)
Comment (youssefali424) [reply packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart]: you are right! but I already tested with the cases uncommented, just uncommented them thank you for noticing.

**Comment by youssefali424** [comment]
@Piinks I think all good now

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
Thank you for the contribution!

File: packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart [hunk 2]
```diff
@@ -18,19 +18,24 @@ void main() {
     ScrollController controller, {
     Axis axis = Axis.vertical,
     bool reverse = false,
+    TextDirection textDirection = TextDirection.ltr,
   }) {
     final AxisDirection axisDirection;
```
Comment (a-wallen) [inline packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart:23]: It's possible that someone will add another case to the switch statement, but not assign `axisDirection` a value. To make sure that it always has a value before it's used you might want to make this a `late final` variable.

**Review by a-wallen — COMMENTED** [review]
Besides one minor nit. LGTM.

File: packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart (hunk 2 above)
Comment (youssefali424) [reply packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart:23]: I don't fully understand, I think you might made a mistake there, it is actually the reverse currently no one can add a switch case without assigning `axisDirection` if I add `late` the compiler won't complain if `axisDirection` isn't assigned
@a-wallen

Comment (a-wallen) [reply packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart:23]: If it's late, but the value is null, the program will throw a runtime error. But you are right, that it's probably better that this is a compile-time error and not a runtime error. 

Thanks for the clarification.

**Review by a-wallen — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart (hunk 2 above)
Comment (youssefali424) [reply packages/flutter/test/widgets/overscroll_stretch_indicator_test.dart:23]: Thank you.

**Comment by youssefali424** [comment]
@Piinks seems google testing is pending (or failed for some reason) but can't detect the issue

**Comment by Piinks** [comment]
It is stuck, thanks for flagging! Merging manually.

**Pull request merged**